### ISSUE TITLE
Apple Getting Out Of The Board

### ISSUE THREAD

**Issue by edgar-ta** [edgar-ta/snake-game#1]
When generating a new apple, this one chooses locations which are moved by the blocks of the snake, and it gets out of the board

**Comment by edgar-ta** [comment]
For solving this, I thought about keeping the same positions for the n<sup>2</sup> blocks within the board while setting the dimension of it to 2n, this way, giving enough space (4n<sup>2</sup> blocks) for the overflow caused by the snake blocks.

The last part would be the css stylings so that the overflow doesn't show; I'm not sure, but I think I could set the grid of the board so that the last n columns and rows have width and height of zero, respectively.

I will work on it later this evening.

**Comment by edgar-ta** [comment]
Turns out the problem wasn't the blocks pushing each
other (that doesn't happen once they get their column
and row set) but the positions the board had.

When I looped to create the positions, I used zero-based
indeces that then I pushed to the positions array.

The grid column/row property uses one-based indeces, so,
sometimes when I grabbed a position from the board, the
apple got a 0 coordinate, which was invalid and changed
to `auto` at the moment of setting the column/row.

Finally, the `auto` value simply made the apple go
at the end of the grid, causing it to overflow.

How I fixed it? Adding 1 to the coordinates pushed to
the positions array.
